cpu/lm4f120/periph/uart.c

```diff
@@ -26,125 +26,151 @@
 #include "periph/uart.h"
 #include "periph_conf.h"
 
+
+/**
+ * @brief UART module size
+ */
+ #define UART_MOD_SIZE  (UART1_BASE - UART0_BASE)
+
+ /**
+  * @brief UART base depending on dev
+  */
+ #define GET_UART_BASE(x) (UART0_BASE + (x * UART_MOD_SIZE))
+
+ /**
+  * @brief UART_IM_R register address depending on dev
+  */
+ #define GET_UART_IM_R(x) (&UART0_IM_R + (x * UART_MOD_SIZE))
+
 /**
  * @brief UART device configurations
  */
 static uart_isr_ctx_t config[UART_NUMOF];
 
-static int init_base(uart_t uart, uint32_t baudrate);
-
 /**
  * Configuring the UART console
  */
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
+    assert(uart < UART_NUMOF);
     /* Check the arguments */
-    assert(uart == 0);
+
+    /* Derive UART parameters based off dev */
+    unsigned long uart_base = GET_UART_BASE(uart_config[uart].dev);
```

Comment:
here and below: please use just "unsigned" instead of "unsigned long"

Reply:
(I'm just realizing that the file looked quite ugly before, so maybe it makes sense to stick with the file's style and fix later)

Reply:
Ya, that is still deriving a 32 bit vendor number UARTx_BASE so I think I need to leave it unsigned long.  I think we are planning to discuss an overhaul and standardization of the periph configs.  I can clean everything later.